packages/markdown/remark/src/remark-shiki.ts

```diff
@@ -33,19 +33,25 @@ export interface ShikiConfig {
 /**
  * getHighlighter() is the most expensive step of Shiki. Instead of calling it on every page,
  * cache it here as much as possible. Make sure that your highlighters can be cached, state-free.
+ * We make this async, so that multiple calls to parse markdown still share the same highlighter.
  */
-const highlighterCache = new Map<string, shiki.Highlighter>();
+const highlighterCacheAsync = new Map<string, Promise<shiki.Highlighter>>();
```

Comment:
Always makes me nervous to cache promises, but this makes sense to me.

Reply:
agreed, see note below about wanting to refactor this post v1.0 beta (for perf but also to get rid of some edge case bugs related to this)